include/RAJA/exec-cuda/forall_cuda.hxx

```diff
@@ -205,7 +266,9 @@ RAJA_INLINE void forall(IndexSet::ExecPolicy<seq_segit, cuda_exec<BLOCK_SIZE>>,
   }  // iterate over segments of index set
 
   cudaErrchk(cudaPeekAtLastError());
-  cudaErrchk(cudaDeviceSynchronize());
+  if (!Async) {
```

Comment:
Would it be better (less error-prone) to bury the conditional and the line above it in macro?